src/shared/sam/cli/samCliDetection.ts

```diff
@@ -56,7 +57,9 @@ export async function detectSamCli(opt: { showMessage: boolean | undefined }): P
             }
         }
 
-        recordSamDetect({ result: currentsamCliLocation === undefined ? 'Failed' : 'Succeeded' })
+        if (!opt.passive) {
```

Comment:
Is it reasonable to add tests around this new conditional, to verify that recordXxx is/isn't being called?

Reply:
Added a test for the login case. `detectSamCli` is not currently covered by any tests, but I verified it manually.